File: CodingQuestions/RedBook/src/main/java/RedBook/RedBook/Entity/Post.java
package RedBook.RedBook.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "post",uniqueConstraints = @UniqueConstraint(columnNames = "title"))
@Getter
@Setter
public class Post {

@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
private long id;

@Column(name = "title",nullable = false)
private String title;

@Column(name = "description",nullable = false)
private String description;

@Column(name = "content",nullable = false)
private  String content;

@CreationTimestamp
private LocalDateTime createTime;

@UpdateTimestamp
private LocalDateTime updateTime;




}

File: MavenProject/KotlinRedbook/src/main/kotlin/com/chuwa/KotlinRedbook/Entity/Post.kt
package com.chuwa.KotlinRedbook.Entity

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Entity
@Table(name = "posts")
data class Post(
                @Column(name = "title", nullable = false)
                var title: String,

                @Column(name = "content", nullable = true)
                var content : String?
        ) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @CreationTimestamp
    var createDateTime : LocalDateTime = LocalDateTime.now()
}